Dynamic derivations RFC 92 (#4628)
NixOS/nix: doc/manual/src/protocols/derivation-aterm.md

> For historical reasons, [derivations](@docroot@/glossary.md#gloss-store-derivation) are stored on-disk in [ATerm](https://homepages.cwi.nl/~daybuild/daily-books/technology/aterm-guide/aterm-guide.html) format.

Reviewer: ```suggestion
# ATerm Derivations

For historical reasons, [derivations](@docroot@/glossary.md#gloss-store-derivation) are stored on-disk in [ATerm](https://homepages.cwi.nl/~daybuild/daily-books/technology/aterm-guide/aterm-guide.html) format.
```

> Derivations either serialized in one of the following formats:

Reviewer: ```suggestion
Derivations are serialised in one of the following formats:
```

We try to use BE consistently.

Author: Ah and what I wrote was just pain ungrammatical too whoops.

- ```
  Derive(...)
  ```

  For all stable derivations.

- ```
  Derive(<version-string>, ...)
  ```

>   The only `version-string`s that are in use today are for [exerimental features]((@docroot@/contributing/experimental-features.md):

Reviewer: ```suggestion
  The only `version-string`s that are in use today are for [exerimental features](@docroot@/contributing/experimental-features.md):
```

For historical reasons, [derivations](@docroot@/glossary.md#gloss-store-derivation) are stored on-disk in [ATerm](https://homepages.cwi.nl/~daybuild/daily-books/technology/aterm-guide/aterm-guide.html) format.

Derivations are serialised in one of the following formats:

- ```
  Derive(...)
  ```

  For all stable derivations.

- ```

>   Derive(<version-string>, ...)

Reviewer 2: ```suggestion
  DrvWithVersion(<version-string>, ...)
```
